src/asyncresults.jl

```diff
@@ -158,9 +157,10 @@ function _cancel(jl_conn::Connection)
         errbuf = zeros(UInt8, errbuf_size)
         success = libpq_c.PQcancel(cancel_ptr, pointer(errbuf), errbuf_size) == 1
         if !success
-            warn(LOGGER, Errors.JLConnectionError(
-                "Failed cancelling query: $(String(errbuf))"
-            ))
+            warn(
+                LOGGER,
```

Comment:
This is my least favourite thing about the style guide but I don't know how to adequately specify the behaviour I use and like here.

Reply:
this is why is favour  `do`-form for logging 😂 

like 
```julia
warn(LOGGER) do
    "looooooooooooong msg wwwwwwwwwwwwwwwwwwwwwwwwooo"
end
```
to avoid
```julia
warn(
     LOGGER,
    "looooooooooooong msg wwwwwwwwwwwwwwwwwwwwwwwwooo"
)
```

Reply:
I wish I knew more about the impact of more anonymous functions on the compiler.